src/main/java/net/minecraftforge/common/extensions/IForgeTagBuilder.java

```diff
@@ -67,4 +67,82 @@
     {
         return getBuilder().add(new ITag.OptionalTagEntry(location));
     }
+    
+    /**
+     * Adds a registry entry to the tag json's remove list. Callable during datageneration.
+     * @param element The entry to remove
+     * @return The builder for chaining
+     */
+    default TagsProvider.Builder<T> removeRegistryEntry(final IForgeRegistryEntry<T> entry)
+    {
+        return removeElementByID(entry.getRegistryName());
+    }
+    
+    /**
+     * Adds multiple registry entries to the tag json's remove list. Callable during datageneration.
+     * @param entries The entries to remove
+     * @return The builder for chaining
+     */
+    @SuppressWarnings("unchecked")
+    default TagsProvider.Builder<T> removeRegistryEntries(final IForgeRegistryEntry<T>...entries)
+    {
+        for (IForgeRegistryEntry<T> entry : entries)
+        {
+            this.removeRegistryEntry(entry);
+        }
+        return getBuilder();
+    }
+    
+    /**
+     * Adds a single element's ID to the tag json's remove list. Callable during datageneration.
+     * @param location The ID of the element to remove
+     * @return The builder for chaining
+     */
+    default TagsProvider.Builder<T> removeElementByID(final ResourceLocation location)
```

Comment:
Use same naming scheme as in 1.17 to reduce upgrading or downgrading complexity.